rasterio/_base.pyx

```diff
@@ -819,17 +819,62 @@ cdef class DatasetBase(object):
         num_items = CSLCount(metadata)
         return dict(metadata[i].split('=', 1) for i in range(num_items))
 
+
+    def get_metadata_item(self, bidx, ns, dm=None, ovr=None):
+        """Returns metadata item
+
+        Parameters
+        ----------
+        bidx: int
+            Band index, starting with 1.
+        name: str
+            The key for the metadata item to fetch.
+        domain: str
+            The domain to fetch for.
+        ovr: int
+            Overview level
+
+        Returns
+        -------
+        str
+        """
+        cdef GDALMajorObjectH band = NULL
+        cdef GDALMajorObjectH obj = NULL
+        cdef char *value = NULL
+        cdef const char *name = NULL
+        cdef const char *domain = NULL
+
+        ns = ns.encode('utf-8')
+        name = ns
+
+        if dm:
+            dm = dm.encode('utf-8')
+            domain = dm
+
+        band = self.band(bidx)
+        if ovr:
+            obj = GDALGetOverview(band, ovr)
+        else:
+            obj = band
+
+        value = GDALGetMetadataItem(obj, name, domain)
+        if value == NULL:
+            return None
+        else:
+            return value
```

Comment:
@vincentsarago would you be willing to try modifying the dataset's `tag()` method instead? We could add an `ovr` keyword argument and then at https://github.com/mapbox/rasterio/blob/master/rasterio/_base.pyx#L811, add a 

```
            if ovr:
                obj = GDALGetOverview(obj, ovr)
```

Reply:
Problem here is you cannot you `GetMetadata` with `TIFF` domain as explained in https://trac.osgeo.org/gdal/ticket/6961 and that's why the main motivation for adding this method

Reply:
Oh, right. Good thing "sgillies" asked that question 9 months ago!